### ISSUE TITLE
Is there any built-in data augmentation function?

### ISSUE THREAD

**Issue by LeeYongHyeok** [KinWaiCheuk/nnAudio#49]
Hi, I'm so impressed by your wonderful project.
But, I want to know how can i augment the training data (ex. SNR control, time-stretch, speed perturbation, volume or pitch control, specaugment ...)
In the torchaudio, there are built-in parameter for the data transformations. [https://pytorch.org/tutorials/beginner/audio_preprocessing_tutorial.html#transformations](url)
Is there any built-in function or parameter for the data augmentation?

**Comment by KinWaiCheuk** [comment]
Hi YongHyeok, thanks for you question. Unfortunately, there is no build in data augmentation in nnAudio at the moment. If you have the augmentation function, however, you can apply it to the PyTorch tensor returned by nnAudio.

**Comment by LeeYongHyeok** [comment]
@KinWaiCheuk 
Thanks for your detailed answer. Do you have the to-do plan about these function?

**Comment by KinWaiCheuk** [comment]
Yes, I do wish to expand nnAudio by adding more features to it (data augmentation is one of the items on my mind). But we are lack of manpower at the moment (me and my current collaborator are occupied with improving the GriffinLim reconstruction accuracy).

If you know anyone who are interested in contributing, please let me know, it would be a great help for us!

**Comment by hbredin** [comment]
Regarding audio augmentation, you might want to join forces with https://github.com/asteroid-team/torch-audiomentations.

**Comment by KinWaiCheuk** [comment]
> Regarding audio augmentation, you might want to join forces with https://github.com/asteroid-team/torch-audiomentations.

Wow, it is a really awesome project! If it is the case, I think a build-in data augmentation is not necessary for nnAudio? Since you can always keep one line of code as audio-domain augmentation, another line of code for nnAudio, e.g.

```python
import torch
from torch_audiomentations import Gain
from nnAudio import Spectrogram
class MyModel():
    def __init___(args):
        self.augmentation_layer = Gain(min_gain_in_db=-15.0, max_gain_in_db=5.0, p=0.5)
        self.nnaudio_layer = Spectrogram.STFT()
        self.neuralnet = SomeNeuralNet()

    def forward(x):
        x = self.augmentation_layer(x)
        x = self.nnaudio_layer(x)
        x = self.neuralnet(x)
        return x

``` 

Is there any augmentation that cannot be done in this way?

**Comment by hbredin** [comment]
I am sure that someone will eventualy come up with a new augmentation technique that must be done in a different way but I don't know of any for now.

**Comment by fdroessler** [comment]
There is also Kornia, I guess that could be used as long as they provide the necessary Augmentations:
https://kornia.readthedocs.io/en/latest/tutorials/data_augmentation.html
and: https://arxiv.org/pdf/2011.09832.pdf
